pkg/controller/lifecyclehandler/handler.go

```diff
@@ -197,20 +197,22 @@ func reasonForUnresolvableDeps(err error) (string, error) {
 
 func (r *LifecycleHandler) EnsureFinalizers(ctx context.Context, original, resource *k8s.Resource, finalizers ...string) error {
 	if !k8s.EnsureFinalizers(resource, finalizers...) {
-		u, err := original.MarshalAsUnstructured()
+		uo, err := original.MarshalAsUnstructured()
 		if err != nil {
 			return err
 		}
-		copy, err := k8s.NewResource(u)
+		originalCopy, err := k8s.NewResource(uo)
 		if err != nil {
 			return err
 		}
-		if !k8s.EnsureFinalizers(copy, finalizers...) {
-			if err := r.updateAPIServer(ctx, copy); err != nil {
+		if !k8s.EnsureFinalizers(originalCopy, finalizers...) {
```

Comment:
I'm confused by what's going on here (or why we have to change it).  Maybe the comment should explain what happens if we don't do this i.e. why are we doing this syncing.

(I suspect the "full" answer is that we should avoid having multiple copies of the Resource, and we should just use `client.Object` - often *unstructured.Unstructured.  But I do think that's a bigger change so I understand not making it here.  I think if we applied the finalizers first - which is more conventional - we could make this change safely)

Reply:
@justinsb do you consider EnsureFinalizers() a blocker? I think this may not be a super quick fix and can be addressed separately so created b/323998236 to track it. Right now I'd prefer prioritizing the `state-into-spec` namespace-level override feature.